Update `about-the-website-development-team.md`

vraer opened the pull request:
### Summary

This PR migrates the About the Website Development Team page from the GitHub wiki to the new mkdocs site.
- **Location in GitHub wiki**: https://github.com/hackforla/website/wiki/About-the-Website-Development-Team
- **New location in website-wiki**: `docs/roles/dev/about-the-website-development-team.md`

### Changes

- Added mermaid diagram to visualize skill progression pathway
- Included note box to highlight exceptions can be discussed
- Formatted content for improved readability and consistency

Please review and let me know if any details need to be modified or expanded on.

---

### Screenshots

<details open><summary><b>Edit</b></summary>

![about-the-website-dev-team_edit](https://github.com/hackforla/website-wiki/assets/124115753/4ad2983e-3f6f-420f-abb0-e42c5da92e8b)

</details> 
<details><summary><b>Original</b></summary>

![about-the-website-dev-team_original](https://github.com/hackforla/website-wiki/assets/124115753/5dd7c15c-f967-4c77-9b73-72d7daf4b47a)

</details>

vraer committed:
Update formatting and issue progression section

vraer commented:
@JessicaLucindaCheng @roslynwythe

I've come across a few open issues related to content duplication (see below). As this PR is being considered, should we also think about instances where we can instead link back to a main source? Like meetings/schedule info or things that don't require additional context or explanation.

<details><summary>Examples of related issues</summary>

- https://github.com/hackforla/website/issues/4918
- https://github.com/hackforla/website/issues/2740
- https://github.com/hackforla/website/issues/2746
- https://github.com/hackforla/website/issues/4543

</details> 

![repeat-info](https://github.com/hackforla/website-wiki/assets/124115753/dede4642-6723-4a95-8ef0-ebedcf3d76c1)

To get this started, we could identify/review topics covered on high-traffic pages:

- [Hack for LA website](https://www.hackforla.org/projects/website) and [Getting Started page](https://www.hackforla.org/getting-started)
- `CONTRIBUTING.md`
- Pages linked in issue/PR templates or mentioned in Slack
- Pages referenced in issues

Thoughts?

JessicaLucindaCheng commented:
Review ETA
- Availability: 2 hours
- ETA: Thursday, October 26, 2023

JessicaLucindaCheng commented:
@vraer 

> I've come across a few open issues related to content duplication (see below). As this PR is being considered, should we also think about instances where we can instead link back to a main source? Like meetings/schedule info or things that don't require additional context or explanation.

I think you make a good point about duplicate content and that it is better to link to one source. I will keep that in mind for #2746 (the issue I am assigned to). Also, I'll see if there is anything on the "About the Website Development Team" that we can have links to instead of repeating on the page when I review your pr.

I can't comment on the particulars of the other issues you listed but in general, I think not repeating content is a good idea. Since @roslynwythe is the current Tech Lead, I'll let her answer your other questions and provide guidance on what she thinks should be worked on next.

JessicaLucindaCheng commented on docs/roles/dev/about-the-website-development-team.md:
```diff
@@ -0,0 +1,70 @@
+- **Mandatory**: Attend at least one of the following meetings each week:
+    - All team weekly meeting (UX/UR, Development, Product) — Sundays 10am-12 noon Pacific Time.
+    - Developer (front-end/back-end) weekly team meeting — Tuesdays 7-8 pm Pacific Time.
+- **Optional**: Developer Office Hours meeting — Thursdays 7-8 pm Pacific Time.
```
To match the content in [2.1 Hack for LA Contributor expectations](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#21-hack-for-la-contributor-expectations) in CONTRIBUTING.md, remove `**Optional**:` and indent the line so it is the same amount of indentation as the other meeting times.

JessicaLucindaCheng reviewed (changes requested):
@vraer Good job refining and rewording the original content! I think it is a lot better.

See my in-line change requests. Thanks!

JessicaLucindaCheng commented on docs/roles/dev/about-the-website-development-team.md:
```diff
@@ -0,0 +1,70 @@
+
+### Communication
+
+- **Announcements**: If you plan to take a week off or go on a longer vacation, notify the tech leadership team. This can be done through a Slack message in the `hfla-site` channel and an @ mention in the comment section of the issue you're working on.
```
I think `Announcements` is a little confusing for me because I think of like making announcements. Maybe instead of `Announcements` it could be `Time Off` or something like that.

JessicaLucindaCheng commented:
@vraer Leaving this as only as suggestions for future versions of this page and not necessary to implement for this pr:
- _Team Levels_ section could probably go on the _Joining the hackforla org website team_ wiki page. 
  - The wording would need to be generalized, so `Development Team` and `new developers` could be changed to `Website Team` and `new members`, respectively 
- _Time Commitment_, _Communication_, _Skill Progression_, and probably the line "Work on one issue at a time" could probably go on the [_Joining the hackforla org website team_](https://hackforla.github.io/website-wiki/pending-audit/onboarding/joining-the-hackforla-org-website-team/#lets-go) wiki page because I think these requirements apply to all members and not just devs.
  - The info could probably go in section [_2.1 Hack for LA_](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#21-hack-for-la-contributor-expectations) in _CONTRIBUTING.md_ as well instead of the wiki. However, I think it makes more sense to put the content in the wiki and link it in section 2.1.
- If a line was added to the "Let's go!" section on [_Joining the hackforla org website team_](https://hackforla.github.io/website-wiki/pending-audit/onboarding/joining-the-hackforla-org-website-team/#lets-go) wiki page saying something like, "Please read the [Website Team's](https://www.hackforla.org/projects/website)" information page to learn about the team, including tools, technologies, and languages we use and our meeting times", I think the _Technologies Used_ section could be omitted. 
- Info for _Meetings_ section, and "Wait until your pull request is merged before working on another issue." could probably be moved to the _Web Developer_ wiki page. 
  - Though, a better option might be to link to the info in section [_2.1 Hack for LA_](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#21-hack-for-la-contributor-expectations) in _CONTRIBUTING.md_ in the _Web Developer_ wiki page because the meetings and the info about prs are already in section 2.1.

JessicaLucindaCheng reviewed (changes requested):
@vraer One more in-line change request.

JessicaLucindaCheng commented:
@vraer Leaving this as only as suggestions for future versions of this page and not necessary to implement for this pr:
- _Team Levels_ section could probably go on the _Joining the hackforla org website team_ wiki page. 
  - The wording would need to be generalized, so `Development Team` and `new developers` could be changed to `Website Team` and `new members`, respectively 
- _Time Commitment_, _Communication_, _Skill Progression_, and probably the line "Work on one issue at a time" could probably go on the [_Joining the hackforla org website team_](https://hackforla.github.io/website-wiki/pending-audit/onboarding/joining-the-hackforla-org-website-team/#lets-go) wiki page because I think these requirements apply to all members and not just devs.
  - The info could probably go in section [_2.1 Hack for LA_](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#21-hack-for-la-contributor-expectations) in _CONTRIBUTING.md_ as well instead of the wiki. However, I think it makes more sense to put the content in the wiki and link it in section 2.1.
- If a line was added to the "Let's go!" section on [_Joining the hackforla org website team_](https://hackforla.github.io/website-wiki/pending-audit/onboarding/joining-the-hackforla-org-website-team/#lets-go) wiki page saying something like, "Please read the [Website Team's](https://www.hackforla.org/projects/website)" information page to learn about the team, including tools, technologies, and languages we use and our meeting times", I think the _Technologies Used_ section could be omitted. 
- Info for _Meetings_ section, and "Wait until your pull request is merged before working on another issue." could probably be moved to the _Web Developer_ wiki page. 
  - Though, a better option might be to link to the info in section [_2.1 Hack for LA_](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#21-hack-for-la-contributor-expectations) in _CONTRIBUTING.md_ in the _Web Developer_ wiki page because the meetings and the info about prs are already in section 2.1.

vraer committed:
Update about-the-website-development-team.md

- Remove 'Optional' from developer office hours meeting 
- Change 'Announcements' to 'Time Off'

JessicaLucindaCheng commented:
Review ETA
- Availability: 2 hours
- ETA: Sunday, October 29, 2023 or Monday, October 30, 2023

JessicaLucindaCheng reviewed (approved):
@vraer LGTM!
